### PR TITLE
Policy Pack for Azure CIS v2.0 Enforcement for Section 9 - AppService.

### PR DESCRIPTION
**Pull request by bigdatasourav**
- [x] Control - 9.1, 9.2, 9.3, 9.5, 9,6, 9.7, 9.8, 9.9, 9.10
- [ ] Control - 9.4  (Need to merge the custom control PR)
- [ ] Control - 9.11 (N/A)

image

image

image

image

image

### CODE REVIEW

File: cis_enforcements/azure/cis_v200/section9/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+---
+categories: ["cis"]
+---
+
+# Enforce Azure CIS v2.0.0 - Section 9 - AppService
+
+This section covers security recommendations for Azure AppService.
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/README.md:7]:
```suggestion
# Enforce Azure CIS v2.0.0 - Section 9 - App Service

This section covers security recommendations for Azure App Service.
```

I can see the CIS doc mentions `AppService`. But let's follow the actual Azure docs and use `App Service`.

File: cis_enforcements/azure/cis_v200/section9/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+> Please note your Terraform state file will then become out of sync and the policy settings should then only be managed in the console.
+> By default, the policies are set to `Check` in the pack's policy settings. To enable automated enforcements, you can switch these policies settings by adding a comment to the `Check` setting and removing the comment from one of the listed enforcement options:
+
+```hcl
+resource "turbot_policy_setting" "azure_appservice_webapp_approved" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/azure-appservice#/policy/types/webAppApproved"
+  note     = "Azure CIS v2.0.0 - Control: 9.1, 9.6, 9,7 and 9.8"
+  value    = "Check: Approved"
+  # value    = "Enforce: Delete unapproved"
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/README.md:84]:
```suggestion
  # value    = "Enforce: Delete unapproved if new"
```

File: cis_enforcements/azure/cis_v200/section9/main.tf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Smart Folder
+resource "turbot_smart_folder" "main" {
+  parent      = "tmod:@turbot/turbot#/"
+  title       = "Azure CIS v2.0.0 - Section 9 - AppService"
+  description = "This section covers security recommendations for Azure AppService."
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/main.tf:5]:
```suggestion
  title       = "Azure CIS v2.0.0 - Section 9 - App Service"
  description = "This section covers security recommendations for Azure App Service."
```

File: cis_enforcements/azure/cis_v200/section9/policies.tf [hunk 4]
```diff
@@ -0,0 +1,172 @@
+# Azure > AppService > WebApp > Approved
+resource "turbot_policy_setting" "azure_appservice_webapp_approved" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/azure-appservice#/policy/types/webAppApproved"
+  note     = "Azure CIS v2.0.0 - Control: 9.1, 9.6, 9,7 and 9.8"
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/policies.tf:5]:
```suggestion
  type     = "tmod:@turbot/azure-appservice#/policy/types/webAppApproved"
  note     = "Azure CIS v2.0.0 - Control: 9.1, 9.6, 9.7 and 9.8"
```

File: cis_enforcements/azure/cis_v200/section9/policies.tf [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+  value    = "Check: Approved"
+  # value    = "Enforce: Delete unapproved"
+}
+
+# Azure > AppService > WebApp > Approved > Custom
+resource "turbot_policy_setting" "azure_appservice_webapp_approved_custom" {
+  resource       = turbot_smart_folder.main.id
+  type           = "tmod:@turbot/azure-appservice#/policy/types/webAppApprovedCustom"
+  note           = "Azure CIS v2.0.0 - Control: 9.1, 9.6, 9,7 and 9.8"
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/policies.tf:14]:
```suggestion
  note           = "Azure CIS v2.0.0 - Control: 9.1, 9.6, 9.7 and 9.8"
```

File: cis_enforcements/azure/cis_v200/section9/policies.tf [hunk 6]
```diff
@@ -0,0 +1,172 @@
+# Azure > AppService > WebApp > Approved
+resource "turbot_policy_setting" "azure_appservice_webapp_approved" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/azure-appservice#/policy/types/webAppApproved"
+  note     = "Azure CIS v2.0.0 - Control: 9.1, 9.6, 9,7 and 9.8"
+  value    = "Check: Approved"
+  # value    = "Enforce: Delete unapproved"
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/policies.tf:7]:
```suggestion
  # value    = "Enforce: Delete unapproved if new"
```

File: cis_enforcements/azure/cis_v200/section9/policies.tf [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+          "message": "App Service authentication is disabled"
+      } -%}
+
+    {%- else -%}
+
+      {%- set data = {
+          "title": "App Service authentication",
+          "result": "Approved",
+          "message": "App Service authentication is enabled"
+        } -%}
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/policies.tf:45]:
```suggestion
    {%- elif $.webApp.appServiceAuth == true -%}

      {%- set data = {
          "title": "App Service Authentication",
          "result": "Approved",
          "message": "App Service authentication is enabled"
        } -%}
        
    {%- else -%}

      {%- set data = {
          "title": "App Service Authentication",
          "result": "Skip",
          "message": "No data for authentication yet"
        } -%}        
```

File: cis_enforcements/azure/cis_v200/section9/policies.tf [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+      } -%}
+
+    {%- endif -%}
+
+    {% set results = results.concat(data) -%}
+    {{ results | json }}
+  EOT
+}
+
+# Azure > AppService > WebApp > HttpsOnly
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/policies.tf:138]:
```suggestion
# Azure > AppService > WebApp > HTTPS Only
```

File: cis_enforcements/azure/cis_v200/section9/policies.tf [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+resource "turbot_policy_setting" "azure_appservice_web_app_https_only" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/azure-appservice#/policy/types/webAppHttpsOnly"
+  note     = "Azure CIS v2.0.0 - Control: 9.2"
+  value    = "Check: Enabled"
+  # value    = "Enforce: Enabled"
+}
+
+# Azure > AppService > WebApp > MinimumTlsVersion
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/policies.tf:147]:
```suggestion
# Azure > AppService > WebApp > Minimum TLS Version
```

File: cis_enforcements/azure/cis_v200/section9/policies.tf [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+resource "turbot_policy_setting" "azure_appservice_web_app_minimum_tls_version" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/azure-appservice#/policy/types/webAppMinimumTlsVersion"
+  note     = "Azure CIS v2.0.0 - Control: 9.3"
+  value    = "Check: TLS 1.2"
+  # value    = "Enforce: TLS 1.2"
+}
+
+# Azure > AppService > WebApp > Http20Enabled
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/policies.tf:156]:
```suggestion
# Azure > AppService > WebApp > HTTP 2.0 Enabled
```

File: cis_enforcements/azure/cis_v200/section9/policies.tf [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+resource "turbot_policy_setting" "azure_appservice_web_app_http20_enabled" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/azure-appservice#/policy/types/webAppHttp20Enabled"
+  note     = "Azure CIS v2.0.0 - Control: 9.9"
+  value    = "Check: Enabled"
+  # value    = "Enforce: Enabled"
+}
+
+# Azure > AppService > WebApp > FtpsState
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/policies.tf:165]:
```suggestion
# Azure > AppService > WebApp > FTPS State
```

File: cis_enforcements/azure/cis_v200/section9/README.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+}
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section9/README.md:85]:
```suggestion
}

resource "turbot_policy_setting" "azure_appservice_web_app_https_only" {
  resource = turbot_smart_folder.main.id
  type         = "tmod:@turbot/azure-appservice#/policy/types/webAppHttpsOnly"
  note         = "Azure CIS v2.0.0 - Control: 9.2"
  # value        = "Check: Enabled"
  value        = "Enforce: Enabled"
}

resource "turbot_policy_setting" "azure_appservice_web_app_minimum_tls_version" {
  resource = turbot_smart_folder.main.id
  type         = "tmod:@turbot/azure-appservice#/policy/types/webAppMinimumTlsVersion"
  note         = "Azure CIS v2.0.0 - Control: 9.3"
  # value        = "Check: TLS 1.2"
  value        = "Enforce: TLS 1.2"
}

resource "turbot_policy_setting" "azure_appservice_web_app_http20_enabled" {
  resource = turbot_smart_folder.main.id
  type         = "tmod:@turbot/azure-appservice#/policy/types/webAppHttp20Enabled"
  note         = "Azure CIS v2.0.0 - Control: 9.9"
  # value        = "Check: Enabled"
  value        = "Enforce: Enabled"
}

resource "turbot_policy_setting" "azure_appservice_web_app_ftps_state" {
  resource = turbot_smart_folder.main.id
  type         = "tmod:@turbot/azure-appservice#/policy/types/webAppFtpsState"
  note         = "Azure CIS v2.0.0 - Control: 9.10"
  # value        = "Check: FTPS only"
  value        = "Enforce: FTPS only"
}
```

File: cis_enforcements/azure/cis_v200/section9/policies.tf (hunk 7 above)
Comment (bigdatasourav) [reply cis_enforcements/azure/cis_v200/section9/policies.tf:45]: The value of `authSettings.enabled` will be either true or false. So `no data` condition will not occur.

**Review by SumitPopat — APPROVED** [review]

**Review by SumitPopat — APPROVED** [review]

**Pull request merged**